### PR TITLE
added :vim command

### PR DESCRIPTION
**Pull request by johngrib**
`:vim` to open file with gui vim (gvim, macvim, etc)
use after `set gvimpath` in .ideavimrc

example :
    `set gvimpath=/usr/local/opt/macvim/bin/mvim`

### CODE REVIEW

File: src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public OpenVimHandler() {
+        super("vi", "m", RANGE_FORBIDDEN | ARGUMENT_FORBIDDEN);
+    }
+
+    public boolean execute(@NotNull Editor editor, @NotNull DataContext context, @NotNull ExCommand cmd) {
+
+        VirtualFile vf = EditorData.getVirtualFile(editor);
+        String filePath = vf.getCanonicalPath();
+
+        StringOption o = (StringOption) Options.getInstance().getOption(Options.GVIM_PATH);
```
Comment (Baldrs) [inline src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java]: Why can't you try to detect system vim in `PATH` if gvimpath is not defined? It should be something simple,  like this example from stackoverflow: 
http://stackoverflow.com/a/32827512/1583006
```java
public static String findExecutableOnPath(String name) {
    for (String dirname : System.getEnv("PATH").split(File.pathSeparator)) {
        File file = new File(dirname, name);
        if (file.isFile() && file.canExecute()) {
            return file.getAbsolutePath();
        }
    }
    throw new AssertionError("should have found the executable");
}
```

Comment (sethk) [inline src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java]: MacOS apps launched by LaunchServices (such as from Finder or the Dock) typically have a very basic `PATH` environment. Don't know about other platforms...

**Comment by zhuyucheng52** [comment]
Place add ex model :g command, this is very useful.

File: src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ */
+public class OpenVimHandler extends CommandHandler {
+    public OpenVimHandler() {
+        super("vi", "m", RANGE_FORBIDDEN | ARGUMENT_FORBIDDEN);
+    }
+
+    public boolean execute(@NotNull Editor editor, @NotNull DataContext context, @NotNull ExCommand cmd) {
+
+        VirtualFile vf = EditorData.getVirtualFile(editor);
+        String filePath = vf.getCanonicalPath();
```
Comment (AlexPl292) [inline src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java]: vf could be null

File: src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        StringOption o = (StringOption) Options.getInstance().getOption(Options.GVIM_PATH);
+        String gVimPath = o.getValue();
+
+        try {
+            String command = gVimPath + " " + filePath;
+            Runtime.getRuntime().exec(command);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (AlexPl292) [inline src/com/maddyhome/idea/vim/ex/handler/OpenVimHandler.java]: Do not suppress exceptions. Please show error message to user, or rethrow it with RuntimeException. In the last case the user will be able to report this error to JetBrains.

**Comment by AlexPl292** [comment]
Hi!
After some discussions, we decided not to include this command in IdeaVim because the original vim has a completely different meaning of this command. If the user wants to have this command, he should define it using `:command`.

**Comment by vlasovskikh** [comment]
I would add that in the original Vim this case is handled by `:!vim %` and we would like to have something similar in IdeaVim so that an interactive command is executed in the Terminal tool window similar to what you've implemented for `:shell`.

**Pull request closed without merging**